feat: add benchmark mode, otel metrics+tracing (#77)
hgaiser/moonshine: README.md

resolve_icons = true
```

> ## Benchmarking

Reviewer: This section is quite long and not necessary for most people. Can you move this to a separate file (maybe in `examples/observability/`?) and reference it here instead?

> Moonshine ships a `bench` subcommand that runs the full host pipeline (compositor + capture + convert + encode) without any Moonlight client connected. Encoded packets are dropped on the floor; per-frame latency samples are aggregated and reported when the run ends. Useful for iterating on driver, power-management, or pipeline changes without a phone in hand.

Reviewer: ```suggestion
Moonshine has a `bench` subcommand that runs the full pipeline (compositor + capture + convert + encode) without any Moonlight client connected. Encoded packets are dropped; per-frame latency samples are aggregated and reported when the run ends. Useful for iterating on driver, power-management, or pipeline changes without needing a client connected.
```
